src/Gui/MainWindow.cpp

```diff
@@ -166,29 +166,84 @@ class DimensionWidget : public QPushButton, WindowParameter
     {
         setFlat(true);
         setText(qApp->translate("Gui::MainWindow", "Dimension"));
+        setToolTip(
+            qApp->translate("Gui::MainWindow", "Select the Unit Schema for the current document"));
         setMinimumWidth(120);
 
-        //create the action buttons
+        // create the action buttons
         auto* menu = new QMenu(this);
         auto* actionGrp = new QActionGroup(menu);
         int num = static_cast<int>(Base::UnitSystem::NumUnitSystemTypes);
+
         for (int i = 0; i < num; i++) {
             QAction* action = menu->addAction(QStringLiteral("UnitSchema%1").arg(i));
             actionGrp->addAction(action);
             action->setCheckable(true);
             action->setData(i);
         }
         QObject::connect(actionGrp, &QActionGroup::triggered, this, [this](QAction* action) {
-            int userSchema = action->data().toInt();
-            setUserSchema(userSchema);
+            bool onceRun = App::GetApplication()
+                               .GetParameterGroupByPath("User parameter:BaseApp/Preferences/Units")
+                               ->GetBool("DoNotShowDocUnitsMsgBox", false);
+            if (!onceRun) {
+                QMessageBox msgBox;
+                QAbstractButton* myYesButton =
+                    msgBox.addButton(tr("OK, but do not show this again"), QMessageBox::YesRole);
```

Comment:
IMO a more standard way of doing this is to use a `QCheckBox` on the dialog that says "Don't show again", and only use the standard OK buttons. See [QMessageBox::setCheckBox](https://doc.qt.io/qt-5/qmessagebox.html#setCheckBox).

Reply:
I'll look at this tomorrow, my goal is always to minimize mouse clicks not because of time saving but RSI, it's only one click saved but imagine if we could do that in 15% of the whole code??

Reply:
Moving to draft pending an update.  @Syres916  please move out of draft when you're ready.